design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+The following commands all mean backup no namespaced resources and all cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+#### some namespaced resources + no cluster resources
+This means backup all resources in namespaces default and kube-system, and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-cluster-resources=false
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in all namespaces, and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=false
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in namespace default and kube-system, and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=false
+```
+
+This means backup all resources except Ingress type resources in all namespaces, and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=Ingress
+	--include-cluster-resources=false
+```
+
+#### some namespaced resources + only related cluster resources
+This means backup all resources in namespaces default and kube-system, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in all namespaces, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in namespaces default and kube-system, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
```

Comment:
Since we're using an explicit list of resources here, to include related cluster resources we'll need to list the types we want to include: --include-cluster-scope-resources=PersistentVolume or whatever. Since include-cluster-resources isn't included, we'll still only include those cluster resources that are relevant to the namespaced resources though, not everything of those types.